frontend/ui/widget/textviewer.lua

```diff
@@ -555,6 +560,26 @@ function TextViewer:reinit()
     self.scroll_text_w:scrollToRatio(ratio, ratio == 0)
 end
 
+function TextViewer:setTextBold(start_pos, len)
+    local end_pos = start_pos + len
+    local text = self.text
+    if self.charlist == nil then
+        self.charlist = util.splitToChars(text)
+    end
+    local charlist = { TextBoxWidget.PTF_HEADER }
+    for i, char in ipairs(self.charlist) do
+        if i == start_pos then
+            table.insert(charlist, TextBoxWidget.PTF_BOLD_START)
+        elseif i == end_pos then
+            table.insert(charlist, TextBoxWidget.PTF_BOLD_END)
+        end
+        table.insert(charlist, char)
+    end
+    self.text = table.concat(charlist)
+    self:reinit()
+    self.text = text
```

Comment:
I guess this strange double `self.text =` assignment is what makes it work multiple times, so might be worth a comment on the last line: `-- restore original text`.
Does it really work when doing multiple searches, when no match, when matching, when moving, when again no match, when matching some pattern overlapping a previous match ?
In no case we would re-use the self.charlist with the PTF_BOLD_START/END in it which would prevent a match of some pattern because they are there in the middle ?

(It also feels it would be better to make bold ALL matches, so one may not need to use Find next - but then, it is quite some different code, and you would need a util.stringSearchAll() :))

Reply:
> Does it really work

Yes it works good.